packages/web3-bzz/src/index.js

```diff
@@ -52,7 +51,7 @@ if (typeof ethereum !== 'undefined' && ethereum.bzz) {
 
 Bzz.prototype.setProvider = function(provider) {
     // is ethereum provider
-    if(_.isObject(provider) && _.isString(provider.bzz)) {
+    if(typeof provider === 'object' && !!provider && typeof provider.bzz === 'string') {
```

Comment:
It will be more efficient to first check that provider is valid (not undefined etc ) and then move on with other checks as its short circuit and. like:

```  if(!!provider && typeof provider === 'object' && typeof provider.bzz === 'string')  ```

Reply:
There are couple of more _same_ improvements possible with !! and && in this PR.